Firmware/ultralcd.cpp

```diff
@@ -6631,44 +6643,50 @@ static void lcd_main_menu()
     MENU_ITEM_SUBMENU_P(_i("Preheat"), lcd_preheat_menu);////MSG_PREHEAT
   }
 
-
-  if(isPrintPaused && saved_printing_type == PRINTING_TYPE_USB)
+  if (mesh_bed_leveling_flag == false && homing_flag == false && !isPrintPaused)
   {
-#ifdef FANCHECK
-      if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))
-          MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);////MSG_RESUME_PRINT c=18
-#else
-      MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);////MSG_RESUME_PRINT c=18
-#endif
+	if (IS_SD_PRINTING)
+	{
+		MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_print);////MSG_PAUSE_PRINT c=18
+	}
+	else if (is_usb_printing)
+	{
+		MENU_ITEM_FUNCTION_P(_T(MSG_PAUSE_PRINT), lcd_pause_usb_print);////MSG_PAUSE_PRINT c=18
+	}
   }
-
-#ifdef SDSUPPORT
-  if (card.cardOK || lcd_commands_type == LcdCommands::Layer1Cal)
+  if(isPrintPaused)
   {
-    if (card.isFileOpen())
-    {
-		if (mesh_bed_leveling_flag == false && homing_flag == false) {
-			if (card.sdprinting)
+	#ifdef FANCHECK
+		if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))
```

Comment:
I'd wrap just the `if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))` into `#ifdef FANCHECK` to avoid having the block body twice in the code:

```
#ifdef FANCHECK
		if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))
#endif
        {
			if (is_usb_printing){
				MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_usb_print);////MSG_RESUME_PRINT c=18
			} else {
				MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);////MSG_RESUME_PRINT c=18
            }
        }
```

Reply:
Pushed change